src/lando/headless_api/models/automation_job.py

```diff
@@ -72,74 +45,27 @@ def processing(self):
         """Return the job details as API status JSON."""
-        return {
-            "job_id": self.id,
-            "status_url": f"{settings.SITE_URL}/api/job/{self.id}",
-            "message": f"Job is in the {self.status} state.",
-            "created_at": self.created_at,
-            "status": self.status,
-            "error": self.error,
-        }
-
-    def transition_status(
-        self,
-        action: JobAction,
-        **kwargs,
-    ):
-        """Change the status and other applicable fields according to actions.
-
-        Args:
-            action (JobAction): the action to take, e.g. "land" or "fail"
-            **kwargs:
-                Additional arguments required by each action, e.g. `message` or
-                `commit_id`.
-        """
-        actions = {
-            JobAction.LAND: {
-                "required_params": ["commit_id"],
-                "status": JobStatus.LANDED,
-            },
-            JobAction.FAIL: {
-                "required_params": ["message"],
-                "status": JobStatus.FAILED,
-            },
-            JobAction.DEFER: {
-                "required_params": ["message"],
-                "status": JobStatus.DEFERRED,
-            },
-            JobAction.CANCEL: {
-                "required_params": [],
-                "status": JobStatus.CANCELLED,
-            },
-        }
-
-        if action not in actions:
-            raise ValueError(f"{action} is not a valid action")
-
-        required_params = actions[action]["required_params"]
-        if sorted(required_params) != sorted(kwargs.keys()):
-            missing_params = required_params - kwargs.keys()
-            raise ValueError(f"Missing {missing_params} params")
-
-        self.status = actions[action]["status"]
-
-        if action in (JobAction.FAIL, JobAction.DEFER):
-            self.error = kwargs["message"]
-
-        if action == JobAction.LAND:
-            self.landed_commit_id = kwargs["commit_id"]
-
-        self.save()
+        job_dict = self.to_dict()
+
+        job_dict["job_id"] = job_dict.pop("id")
+        job_dict["message"] = f"Job is in the {self.status} state."
+        job_dict["status_url"] = f"{settings.SITE_URL}/api/job/{self.id}"
+
+        return job_dict
 
     @classmethod
-    def next_job(cls, repositories: Optional[Iterable[str]] = None) -> Self:
-        """Return the next automation job."""
-        return (
-            cls.objects.filter(status__in=(JobStatus.SUBMITTED, JobStatus.DEFERRED))
-            .filter(target_repo__in=repositories)
-            .order_by("-priority", "created_at")
-            .select_for_update()
-        )
+    def job_queue_query(
+        cls, repositories: Optional[Iterable[str]] = None, **kwargs
+    ) -> QuerySet:
+        """Return a query which selects the queued jobs."""
+        # XXX: If we can include IN_PROGRESS jobs, and sort by status_order, this method
+        # can be deleted in favour of the parent class's.
```

Comment:
@cgsheeh you may be the best placed to know:

I'm pretty sure https://github.com/mozilla-conduit/lando/pull/475/files#diff-857711dd1362a2dcfe7cc7b4a4ae6c412405c88e966aeb33600d517bcdd8354cR231-R238 would work equally well, so we can just remove the override.

Reply:
Yes, I think we can include in progress jobs and remove the override. 👍